File: oa-portal/problems/m55-transaction-router/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

/* initial[i] = {p, c} for gateway i+1.
   ops[j] = {type, a, b}: type 0 is ADD (a=p, b=c), type 1 is REMOVE (a=id),
   type 2 is ROUTE (a=r). Return one answer per ROUTE, in order. */
vector<long long> answerRoutes(const vector<pair<long long,long long>>& initial,
                               const vector<array<long long,3>>& ops) {
    // write your code here

    return vector<long long>();
}

int main() {
    int n, q;
    if (scanf("%d %d", &n, &q) != 2) return 0;
    vector<pair<long long,long long>> initial(n);
    for (auto &g : initial) scanf("%lld %lld", &g.first, &g.second);

    vector<array<long long,3>> ops(q);
    char word[16];
    for (int i = 0; i < q; i++) {
        scanf("%s", word);
        if (word[0] == 'A') { ops[i][0] = 0; scanf("%lld %lld", &ops[i][1], &ops[i][2]); }
        else if (word[1] == 'E') { ops[i][0] = 1; scanf("%lld", &ops[i][1]); ops[i][2] = 0; }
        else { ops[i][0] = 2; scanf("%lld", &ops[i][1]); ops[i][2] = 0; }
    }

    string out;
    for (long long v : answerRoutes(initial, ops)) { out += to_string(v); out += '\n'; }
    fwrite(out.data(), 1, out.size(), stdout);
    return 0;
}

File: oa-portal/problems/m55-transaction-router/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    /* initial[i] = {p, c} for gateway i+1.
       ops[j] = {type, a, b}: type 0 is ADD (a=p, b=c), type 1 is REMOVE (a=id),
       type 2 is ROUTE (a=r). Return one answer per ROUTE, in order. */
    static long[] answerRoutes(long[][] initial, long[][] ops) {
        // write your code here

        return new long[0];
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int n = in.nextInt(), q = in.nextInt();
        long[][] initial = new long[n][2];
        for (int i = 0; i < n; i++) { initial[i][0] = in.nextLong(); initial[i][1] = in.nextLong(); }
        long[][] ops = new long[q][3];
        for (int i = 0; i < q; i++) {
            String w = in.next();
            if (w.charAt(0) == 'A') { ops[i][0] = 0; ops[i][1] = in.nextLong(); ops[i][2] = in.nextLong(); }
            else if (w.charAt(1) == 'E') { ops[i][0] = 1; ops[i][1] = in.nextLong(); }
            else { ops[i][0] = 2; ops[i][1] = in.nextLong(); }
        }

        StringBuilder sb = new StringBuilder();
        for (long v : answerRoutes(initial, ops)) sb.append(v).append('\n');
        System.out.print(sb);
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}
